### ISSUE TITLE
Use .editorconfig and remove per file configuration.

### ISSUE THREAD

**Issue by yurydelendik** [mozilla/pdf.js#6626]
See http://editorconfig.org/

Also we shall remove the following header from all source files:

```
/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
```

The build scripts might be adjusted as well.

**Comment by prometheansacrifice** [comment]
How about the following

```
root = true

[*]
charset = utf-8
end_of_line = lf
indent_size = 2
indent_style = space
insert_final_newline = true
max_line_length = 80
trim_trailing_whitespace = true

[*.md]
max_line_length = 0
trim_trailing_whitespace = false

[COMMIT_EDITMSG]
max_line_length = 0
```

**Comment by yurydelendik** [comment]
instead of `[*]` as shall explicitly list source files

**Comment by prometheansacrifice** [comment]
That's like 42 files. Why explicitly list all source files?

**Comment by yurydelendik** [comment]
I mean [*.{js,jsm,html,css}], not sure why we need to worry about other files

**Comment by timvandermeij** [comment]
Fixed by #6627.
